python/test/unit/language/test_core.py

```diff
@@ -3308,7 +3308,10 @@ def test_dot(M, N, K, num_warps, col_a, col_b, epilogue, input_precision, in_dty
         if in_dtype == 'bfloat16':
             pytest.xfail("bfloat16 is not supported in the interpreter")
     else:
-        if not is_hip() and (M < 16 or N < 16 or K < 16):
+        if is_xpu():
+            if (M < 8 or N < 16 or (K < 16 and in_dtype == 'float16') or (K < 8 and in_dtype == 'float32')):
+                pytest.xfail("XPU: small dots are not supported")
+        elif not is_hip() and (M < 16 or N < 16 or K < 16):
```

Comment:
```suggestion
        if not is_hip() and (M < 16 or N < 16 or K < 16):
```
Use `if not is_hip()` to align with `if is_xpu` and `if is_cuda()`

Reply:
if I do that, then cases when M >= 8 and M < 16 would be skipped, but we want to run them.

Reply:
OK, I see.